### PR TITLE
[backend] rework directive and add system_token directive #1124

### PR DESCRIPTION
**Pull request by jpkha**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.

We need to set a system_token_directive to be authorized to perform specific actions in XTM server. Use for ingesting manifest but will be also used for Trial.

You can try to call this api endpoint with and without the header :
image

## Related issues

- Related to #1124

# Checklist

## Quality

- [ ] I consider this work complete and ready for review
- [ ] I tested all features and edge cases
- [ ] I refactored code where necessary to improve quality
- [ ] I made sure my code meets development guidelines
- [ ] I performed a self-review of my code

## Testing

- [ ] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/ingest-manifest/ingest-manifest.app.ts [hunk 1]
```diff
@@ -9,11 +9,9 @@ const OpenCTIConnectorsManifest =
   'https://raw.githubusercontent.com/OpenCTI-Platform/connectors/master/manifest.json';
 
 export const IngestManifestApp = {
-  async refreshOpenCTIManifest() {
+  async refreshOpenCTIManifest(context: PortalContext): Promise<boolean> {
     const manifest = await fetchManifest(OpenCTIConnectorsManifest);
-
-    // TODO need to create system_user_context from directive on checking token
-    upsertConnectors(SYSTEM_USER_CONTEXT, extractManifestInformation(manifest));
+    await upsertConnectors(context, extractManifestInformation(manifest));
```
[the remark this answers is not in the document]
Comment (jpkha) [reply apps/portal-api/src/modules/ingest-manifest/ingest-manifest.app.ts]: https://github.com/FiligranHQ/xtm-hub/pull/1128/commits/97563d36c8eafe684e025fad879ff0837d30bdf2#diff-02b279984020f5ad3260d9bcff4bd56dfd66fefa107f224d90617bc2c9bfa602

File: apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts [hunk 2]
```diff
@@ -49,9 +50,10 @@ export const createSystemTokenResolver = (originalResolve) => {
         ...portalContext,
         user: SYSTEM_USER_CONTEXT.user,
       };
-
-      // Execute resolver with system context
-      return originalResolve(source, args, enhancedContext, info);
+      requestContext.set({
```
Comment (carinelebas) [inline apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts]: wdyt about using `requestContext.update`  ?
- it would override the user and context if already set before
- it would not override other fields, if we need to add some later?

**Review by carinelebas — APPROVED** [review]

File: apps/portal-api/src/modules/services/document/document.domain.ts [hunk 3]
```diff
@@ -130,27 +129,26 @@ export const upsertDocumentWithChildren = async <T extends DocumentModel>(
...
-  context: PortalContext,
   doc: T,
   upload: Upload[] | Upload,
   trx: Knex.Transaction
 ) => {
-  const files = await processUploads(upload, context);
+  const { portalContext } = requestContext.require();
+  const files = await processUploads(upload, portalContext);
 
   // Get all existing child image documents
```
Comment (Kroustille) [inline apps/portal-api/src/modules/services/document/document.domain.ts]: Remove this comment, it's no longer relevant with the code below

File: apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  return async function secureResolver(
+    source,
+    args,
+    portalContext: PortalContext,
+    info
+  ) {
+    const hasValidToken = validateSystemToken(portalContext);
+
+    if (!hasValidToken) {
+      throw new Error('Valid system token is required for this operation');
```
Comment (Kroustille) [inline apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts]: Do we need to throw a GraphQL error here ?

Comment (jpkha) [reply apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts]: Yes as we call from a Graphql endpoint

Comment (jpkha) [reply apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts]: but it's not for the front so should I add it to the enum error ?

File: apps/portal-api/src/modules/services/document/document.domain.ts [hunk 5]
```diff
@@ -193,14 +190,16 @@ export const upsertDocument = async <T extends DocumentModel>(
...
     ...omit(documentData, ['parent_document_id', 'labels', ...metadataKeys]),
-    uploader_id: context.user.id,
-    uploader_organization_id: context.user.selected_organization_id,
+    uploader_id: contextUser.id,
+    uploader_organization_id: contextUser.selected_organization_id,
   };
 
   // Upsert on slug
-  const [document] = await db<DocumentModel>(context, 'Document')
+  const [document] = await db<DocumentModel>(portalContext, 'Document')
```
[the remark this answers is not in the document]
Comment (jpkha) [reply apps/portal-api/src/modules/services/document/document.domain.ts]: Will do in another PR

File: apps/portal-api/src/modules/services/document/document.domain.ts (hunk 3 above)
Comment (jpkha) [inline apps/portal-api/src/modules/services/document/document.domain.ts]:
```suggestion
```

File: apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts (hunk 4 above)
Comment (jpkha) [reply apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts]: To hide why it's failing ?
